feature/login/src/main/java/com/epmedu/animeal/login/profile/data/repository/ProfileRepositoryImpl.kt

```diff
@@ -0,0 +1,46 @@
+package com.epmedu.animeal.login.profile.data.repository
+
+import androidx.datastore.core.DataStore
+import androidx.datastore.preferences.core.Preferences
+import androidx.datastore.preferences.core.edit
+import androidx.datastore.preferences.core.stringPreferencesKey
+import com.epmedu.animeal.login.profile.data.model.Profile
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.flowOf
+import kotlinx.coroutines.flow.map
+import javax.inject.Inject
+
+internal class ProfileRepositoryImpl @Inject constructor(
+    private val dataStore: DataStore<Preferences>
+) : ProfileRepository {
+
+    override suspend fun getProfile(): Flow<Profile> {
+        return dataStore.data.map { preferences ->
+            Profile(
+                firstName = preferences[FIRST_NAME] ?: EMPTY_STRING,
+                lastName = preferences[LAST_NAME] ?: EMPTY_STRING,
+                email = preferences[EMAIL] ?: EMPTY_STRING,
+                birthDate = preferences[BIRTH_DATE] ?: EMPTY_STRING,
+            )
+        }
+    }
+
+    override suspend fun saveProfile(profile: Profile): Flow<Unit> {
```

Comment:
do we really need return type `Flow<Unit>`?

Reply:
After saving succesfully should navigate next screen , in orderto detect savign to preferences I returned Flow<Unit>